### DISCUSSION TITLE
ESP-C3-C05 ESP32-C3 4Mb Module -- Invalid image block, can't boot.

### DISCUSSION THREAD

**Discussion by ManfredBartz** [arendst/Tasmota#14973] [Q&A] [answered]
I have erased and programmed the [ESP-C3-C05 ESP32-C3 4Mb](https://templates.blakadder.com/DT-ESP-C05.html) module 
with the image from the same web page.

The erasing and programming appears to work ok, but when I reboot the module I get this on the UART port:
```
ESP-ROM:esp32c3-api1-20210207
Build:Feb  7 2021
rst:0x7 (TG0WDT_SYS_RST),boot:0xd (SPI_FAST_FLASH_BOOT)
Saved PC:0x40047ed2
SPIWP:0xee
mode:DOUT, clock div:1
load:0x3c120020,len:0x366e8
load:0x3fc8d000,len:0x33e0
load:0x40380000,len:0x6520
load:0x42000020,len:0x116e38
Invalid image block, can't boot.
ets_main.c 333
```
Any suggestions?

This is my workflow:
```
mbz:~/esptool/esptool-3.2# esptool.py --chip esp32c3 --port /dev/ttyUSB1 erase_flash
esptool.py v3.2
Serial port /dev/ttyUSB1
Connecting....
Chip is ESP32-C3 (revision 3)
Features: Wi-Fi
Crystal is 40MHz
MAC: 7c:df:a1:63:5c:d4
Uploading stub...
Running stub...
Stub running...
Erasing flash (this may take a while)...
Chip erase completed successfully in 21.9s
Hard resetting via RTS pin...

mbz:~/esptool/esptool-3.2# esptool.py --chip esp32c3 --port /dev/ttyUSB1 write_flash -fm dout 0x0 ~mbz/Downloads/tasmota32c3.bin 
esptool.py v3.2
Serial port /dev/ttyUSB1
Connecting....
Chip is ESP32-C3 (revision 3)
Features: Wi-Fi
Crystal is 40MHz
MAC: 7c:df:a1:63:5c:d4
Uploading stub...
Running stub...
Stub running...
Configuring flash size...
Flash will be erased from 0x00000000 to 0x0015dfff...
Compressed 1431440 bytes to 916302...
Wrote 1431440 bytes (916302 compressed) at 0x00000000 in 81.2 seconds (effective 141.1 kbit/s)...
Hash of data verified.

Leaving...
Hard resetting via RTS pin...

mbz:~/esptool/esptool-3.2# putty
```

**Comment by barbudor** [comment, marked as answer]
ESP32, including C3, are slightly more complex to serial flash. For simple serial flashing use 'tasmota32c3.factory.bin' from https://github.com/tasmota/install/tree/main/firmware/development
Or use the web flasher https://tasmota.github.io/install/

More details on https://tasmota.github.io/docs/ESP32/ although it covers ESP32 and as not been updated for C3, you will see that there are 4 binaries to flash
The 'factory' binary merge all 4 into a single one for ease of operations

**Reply by ManfredBartz** [reply]
Thanks, that worked well. :-)
I have configured it for CCT downlights and it shows up in Home Assistant with the appropriate controls.  Very cool!
I tried to do an OTA firmware update with 'tasmota32c3.factory.bin', but that failed:
![Screenshot from 2022-02-26 11-06-33x0 5](https://user-images.githubusercontent.com/2176796/155819644-3cc4687b-217c-48aa-b04e-21c1c7bb211e.png)
Before I put the lights into the ceiling, I'd like to have this working.  Any hints?

**Reply by ManfredBartz** [reply]
I think I figured it out.  The 'factory' image includes the boot-loader, partition-table, boot-app and the Tasmota binary.
The OTA update only requires the Tasmota binary ('tasmota32c3.bin').

**Reply by barbudor** [reply]
Glad to know you've figured all details 👍
Enjoy Tasmota

**Comment by ferbulous** [comment]
@ManfredBartz  hi, may i know how you got your C05 into boot mode?
I tried setting gpio09 to low by connecting the pin to gnd but it’s not working for me yet
